Lib/uuid.py

```diff
@@ -719,14 +719,37 @@ def uuid5(namespace, name):
     hash = sha1(namespace.bytes + name).digest()
     return UUID(bytes=hash[:16], version=5)
 
+def uuid8(a=None, b=None, c=None):
+    """Generate a UUID from three custom blocks.
+
+    * 'a' is the first 48-bit chunk of the UUID (octets 0-5);
+    * 'b' is the mid 12-bit chunk (octets 6-7);
+    * 'c' is the last 62-bit chunk (octets 8-15).
+
+    When a value is not specified, a pseudo-random value is generated.
+    """
+    if a is None:
+        import random
+        a = random.getrandbits(48)
+    if b is None:
+        import random
+        b = random.getrandbits(12)
+    if c is None:
+        import random
+        c = random.getrandbits(62)
```

Comment:
Does it make sense / is it possible to reject values outside the expected value range? For example, reject negative numbers?

Maybe something like:

```
orig_a = a
a &= (a & 0xffff_ffff_ffff)
if a != orig_a: raise ValueError("...")
```

I don't know. Would it be consistent with other uuid functions?

Reply:
UUIDv1 does not reject them so I wouldn't worry too much about it. v3 and v5 are not based on integral inputs.

Reply:
I'm fine with having the same behavior than uuid1() in this case.